src/clr-angular/data/datagrid/_datagrid.clarity.scss

```diff
@@ -3,266 +3,271 @@
 // The full license information can be found in LICENSE in the root directory of this project.
 
 @include exports('datagrid.clarity') {
-    @include basic-table(".datagrid", ".datagrid-head", ".datagrid-body",
-        ".datagrid-row", ".datagrid-column", ".datagrid-cell");
-
-    $clr-datagrid-icon-size: 0.583333rem;
-    $clr-datagrid-action-arrow-size: 0.25rem;
-    $clr-datagrid-fixed-column-size: (2 * $clr-table-cellpadding) + $clr-table-lineheight;
-    $clr-datagrid-column-handle-tracker-top: -0.5rem;
-    $clr-datagrid-column-separator-expandby: 0.5rem;
-    $clr-datagrid-compact-outline-offset: -0.1666667rem;
+    @include basic-table(".datagrid", ".datagrid-header", ".datagrid-body",
+            ".datagrid-row", ".datagrid-column", ".datagrid-cell");
 
     .datagrid-host {
         display: flex;
         flex-flow: column nowrap;
-    }
-
-    .datagrid-overlay-wrapper {
-        display: flex;
-        flex-direction: row;
-        flex: 0 auto; // Same as flex: initial. It sizes the item based on width/height properties or its content.
+        height: auto;
+        position: relative; // for the .datagrid-spinner to work.
         width: 100%;
-        min-height: 100%; // Must be min to fix Safari bug: height: 100% won't fill flex parent. http://bit.ly/2qAPkha
-        overflow-x: auto;
-        overflow-y: hidden;
-
-        .datagrid-spinner {
-            flex: 0 0 100%;
-            margin-left: -100%;
-            display: flex;
-            justify-content: center;
-            align-items: center;
-            background: $clr-datagrid-loading-background;
-        }
-
-    }
-    .datagrid-scroll-wrapper {
-        display: flex;
-        flex-direction: row;
-        min-width: 100%;
     }
+
     .datagrid {
         display: flex;
-        flex-flow: column nowrap;
+        flex-wrap: wrap;
+        flex: 1 1 auto;
+        width: 100%;
+        margin-top: 0;
```

Comment:
Why did you add a`margin-top: 0` here? Could this be the reason that we see the diffs in the CSS regression images?

Reply:
datagrid inherits from the basic table style that has a margin-top of 1rem. Since the clr-dg-action-bar needs to be 0.5rm above the datagrid I reset it here and added bottom-margin to the action bar. I'm not clear on what the spec is for datagrid top margin and the action-bar.